src/meta/processors/admin/Balancer.cpp

```diff
@@ -580,79 +671,85 @@ int32_t Balancer::acquireLeaders(
-        std::unordered_map<HostAddr, std::vector<PartitionID>>& leaderHostParts,
-        std::unordered_map<PartitionID, std::vector<HostAddr>>& peersMap,
-        std::unordered_set<HostAddr>& activeHosts,
-        HostAddr host,
-        size_t maxLoad,
-        LeaderBalancePlan& plan,
-        GraphSpaceID spaceId) {
+int32_t Balancer::giveupLeaders(HostParts& leaderParts,
+                                PartAllocation& peersMap,
+                                std::unordered_set<HostAddr>& activeHosts,
+                                HostAddr host,
+                                LeaderBalancePlan& plan,
+                                GraphSpaceID spaceId) {
     int taskCount = 0;
-    auto& hostLeaders = leaderHostParts[host];
+    auto& hostLeaders = leaderParts[host];
     // host will try to transfer the extra leaders to other peers
-    for (auto it = hostLeaders.begin(); it != hostLeaders.end(); ) {
+    for (auto it = hostLeaders.begin(); it != hostLeaders.end(); it++) {
         // find the leader of partId
         auto partId = *it;
         auto peers = peersMap[partId];
-        bool transfered = false;
-        for (const auto& peer : peers) {
-            if (host == peer || !activeHosts.count(peer)) {
-                continue;
-            }
-            // If peer can accept this partition leader, than host will transfer to the peer
-            auto& peerLeaders = leaderHostParts[peer];
-            if (peerLeaders.size() < maxLoad) {
-                it = hostLeaders.erase(it);
-                peerLeaders.emplace_back(partId);
-                plan.emplace_back(spaceId, partId, host, peer);
-                LOG(INFO) << "plan trans leader: " << spaceId << " " << partId << " host "
-                          << host.host << ":"
-                          << host.port << " peer "
-                          << peer.host
-                          << ":" << peer.port;
-                ++taskCount;
-                transfered = true;
-                break;
-            }
-        }
-        if (!transfered) {
-            ++it;
+        auto maxLoad = partBounds_[partId].second;
+
+        // leader should move to the peer with lowest loading
+        auto min = std::min_element(peers.begin(), peers.end(),
+                                    [&](const auto &l, const auto &r) -> bool {
+                                        if (host == l || !activeHosts.count(l)) {
+                                            return false;
+                                        }
+                                        return leaderParts[l].size() < leaderParts[r].size();
+                                    });
+        // If peer can accept this partition leader, than host will transfer to the peer
+        // if the space specified with zone, we should check for conflict.
+        auto& peerLeaders = leaderParts[*min];
+        int32_t leaderCount = peerLeaders.size();
+        if (leaderCount < hostBounds_[*min].second) {
```

Comment:
Why compare with `minLoad`, aka `partBounds_[partId].first` in line 689, and in this line was compared with `hostBounds_[*min].second`? I suppose we only one of `partBounds_` and `hostBounds_`